### PR TITLE
feat(device/android/device): add custom mapping for tolino buttons

### LINKED ISSUE #9175: FR: Tolino Vision 5 buttons dont invert based on rotation
**Does your feature request involve difficulty completing a task? Please describe.**
Currently the buttons on a `Tolino Vision 5` do not invert based on document view rotation

**Describe the solution you'd like**
Option to enable invert of tolino's buttons (based on rotation and maybe manually)

**Describe alternatives you've considered**
From what i can tell, there is also currently no button to invert the buttons manually (at least i have not found a setting)

**Additional context**
I am still new to koreader and i use it now on my Tolino Vision 5 (Android 4.4), where koreader is installed through F-Droid, koreader version is `2022.05.1`

I have also already tried the options `Tap and Gestures > Page turns > Invert page turn buttons`, which did not seem to affect the Button page turn

~~Also i have noticed that the manual Rotation is only for the Document Viewer and when going back to the file browser it de-rotates again, is this intentional?~~ found setting "Keep orientation across Views"

PS: is it currently possible to have automatic rotation (preferably asked before rotation) based on the gyroscope? (if available)

Update: from what i can tell, the `Tolino Vision 5` is more or less the same as `Kobo Libra H2O`

### PR DESCRIPTION
**Pull request by hasezoey**
This PR adds button mapping for Tolinos

fixes #9175
closes #9503
~~requires https://github.com/koreader/android-luajit-launcher/pull/381~~
requires https://github.com/koreader/android-luajit-launcher/pull/383

This PR is the result of https://github.com/koreader/koreader/pull/9503#issuecomment-1242992368

Note: i did not change to original PR because that was with a different intend, also tried to add some warmth settings (but i could not get that to work and i dont know how to use JNI to implement it)

- - -
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/9509)

### CODE REVIEW

File: frontend/device/android/device.lua
Comment (pazos) [inline frontend/device/android/device.lua:135]: did you mean `android_eventmap` here?.

File: frontend/device/android/device.lua
Comment (pazos) [inline frontend/device/android/device.lua:144]: and here?

File: frontend/device/android/device.lua
Comment (hasezoey) [reply frontend/device/android/device.lua:135]: i dont know what you mean, the variable is named `event_map` and the file that is imported is `device/android/event_map`, like before the PR

Comment (pazos) [reply frontend/device/android/device.lua:135]: no, i mean your variable is named `event_map` but you're asigning new actions to `android_eventmap`, which is `nil`. You need to use the same variable name everywhere.

Comment (hasezoey) [reply frontend/device/android/device.lua:135]: oh, thats what you meant, i think i forgot to change it

Comment (hasezoey) [reply frontend/device/android/device.lua:135]: fixed with 42135c8b3a460aa4be80877ab0b3b61044b61a9c

File: frontend/device/android/device.lua
Comment (hasezoey) [reply frontend/device/android/device.lua:144]: fixed with 42135c8b3a460aa4be80877ab0b3b61044b61a9c

**Comment by hasezoey** [comment]
forgot to mention this PR also requires https://github.com/koreader/android-luajit-launcher/pull/381, because `android.prop.mode` (`Build.MODEL`/`ro.product.model`) is used

**Comment by pazos** [comment]
@hasezoey: instead of using `android.prop.model` lets require https://github.com/koreader/android-luajit-launcher/pull/381 and use `android.prop.is_tolino` instead.

File: frontend/device/android/device.lua
Comment (pazos) [inline frontend/device/android/device.lua:140]:
```suggestion
    if android.prop.is_tolino then
        -- dpad left/right as page back/forward
        event_map[21] = "LPgBack"
        event_map[22] = "LPgFwd"
    end
```

**Review by pazos — CHANGES_REQUESTED** [review]
Code is ready to go.

Lets wait until the requirements are approved and merged and then please add a commit bumping android-luajit-launcher to master.

**Comment by hasezoey** [comment]
> Lets wait until the requirements are approved and merged and then please add a commit bumping android-luajit-launcher to master.

just to clarify, you mean rebasing in this case instead of just merging?

btw, why to choose to rebase instead of just merging in these cases / for these projects? (just curious)

**Comment by pazos** [comment]
no, I mean
```
cd platform/android/luajit-launcher
git checkout master
git pull origin master
cd -
git add platform/android/luajit-launcher
```

**Comment by hasezoey** [comment]
> Lets wait until the requirements are approved and merged and then please add a commit bumping android-luajit-launcher to master.

should be done with 31249f55637fe95094802867aa1ce904c3709eef

**Review by pazos — APPROVED** [review]
LGTM

**Pull request merged**
